SpaDES4Dummies.Rmd

```diff
@@ -10,116 +10,142 @@ output: html_document
+```{r the_r_way_plot_results_AB}
 abundance <- abundance_model(r = r, Time = Time)
 plot(stack(abundance))
+```
 
-## TEMPERATURE "SIMULATIONS"
-## The temperature simulation model will be almost the same as the vegetation one - remember this is a dummy example ;)
-r <- raster(nrows = 100, ncols = 100, xmn = -50, xmx = 50, ymn = -50, ymx = 50)
+#### TEMPERATURE "SIMULATIONS"
+The temperature simulation model will be similar to the vegetation one - remember this is a dummy example ;)
+```{r the_r_way_simulation_function_TMP}
 temp_model <- function(r, Time) {
   temp_outputs <- list()
   for(t in 1:Time) { 
     temp_outputs[[t]] <- SpaDES.tools::gaussMap(r, scale = 100, var = 0.1) 
   }
   return(temp_outputs)
 }
+```
 
-## Lenght of the simulation (or simply the number of model iterations)
-Time <- 10
-
-## Run the model
+Run the model and plot results (all TEMPERATURE plots together)
+```{r the_r_way_plot_results_TMP}
 temperature <- temp_model(r = r, Time = Time)
 plot(stack(temperature))
+```
 
-## DATA ANALYSIS
+#### DATA ANALYSIS
+Now we analyze if abundance of species and temperature are correlated.  
+First we create the data analysis funtion (a simple linear model):
+```{r data_analysis_model}
 stats_analysis <- function(Data){
   lm1 <- lm(abund ~ temp, data = Data)
-  plot(Data$abund ~ Data$temp, xlab = "Temp.", ylab = "Species abundance")
+  plot(Data$abund ~ Data$temp, xlab = "Temperature", ylab = "Species abundance")
   abline(a = lm1$coefficients["(Intercept)"], b = lm1$coefficients["temp"], lwd = 2, col = "blue")
 }
+```
 
-par_sets <- par(mfrow = c(2, 5))
+Then we create a loop to analyze each plot of our time-series:
+```{r data_analysis, echo=-1}
+par(mfrow = c(2, 4)) # This plots in 2 lines and 4 columns
 for(t in 1:Time){
   outputdata <- data.frame(abund = abundance[[t]][], temp = temperature[[t]][])
   stats_analysis(Data = outputdata)
 }
 ```
 
-```{r, echo = FALSE}
-par(par_sets)
-```
-
-
-### After...
-Let's now solve the same problem using the **`SpaDES`** approach. I like to start by creating a *global.R* script (you can call it whatever you want, though) that will load the **`SpaDES`** libraries and run **`SpaDES`** simulations.
+That's it. You have your model. But what if you need to include new data? You would have to manually change the code for it... 
 
+### AFTER  `SpaDES`...
+Let's now solve the same problem using the **`SpaDES`** approach. I like to start by creating a *global.R* script (you can call it whatever you want, though) that will load the **`SpaDES`** libraries and run **`SpaDES`** simulations. The *global.R* script for this dummy example can be found on the root of SpaDES4Dummies Project. 
 ```{r globalscript}
 suppressMessages(library(SpaDES))  ## should automatically download all packages in the SpaDES family and their dependencies
 
 ## decide where you're working
+mainDir <- 'C:/Users/tmichele/Dropbox/Project SpaDES/SpaDES4Dummies/' # SET YOUR MAIN DIRECTORY HERE
```

Comment:
please replace this for a relative path (instead of the absolute one you've used)